gw_eccentricity/load_data.py

```diff
@@ -1030,6 +1275,100 @@ def reomve_junk_from_nr_data(t, modes_dict, num_orbits_to_remove_as_junk):
+    horizon_filepath: str
+        Path to the `Horizons.h5` file. This file typically can be found in the
+        same directory where the waveform files are located in the SXS catalog.
+
+    deltaTOverM: float, default=1
+        Time step in dimensionless units. It is used to interpolate the data
+        since the NR data in `Horizons.h5` are not stored at uniform time grid.
+
+    Returns
+    -------
+    t_clean: array-like
+        Time array corresponding to clean NR data.
+
+    modes_dict_clean: dict
+        modes_dict with `num_orbits_to_remove_as_junk` orbits removed from the
+        begining of modes array.
+    """
+    num_orbits_duration = get_num_orbits_duration_from_horizon_data(
+        horizon_filepath, num_orbits_to_remove_as_junk)
+    idx_junk = np.argmin(np.abs(t - (t[0] + num_orbits_duration)))
+    t_clean = t[idx_junk:]
+    modes_dict_clean = {}
+    for key in modes_dict:
+        modes_dict_clean[key] = modes_dict[key][idx_junk:]
+
+    return t_clean, modes_dict_clean
+
+
+def get_num_orbits_duration_from_horizon_data(horizon_filepath, num_orbits,
+                                              deltaTOverM=1):
+    """Get the duration of `num_orbits` from phase data in `Horizons.h5`.
+
+    Parameters
+    ----------
+    horizon_filepath: str
+        Path to the `Horizons.h5` file. This file typically can be found in the
+        same directory where the waveform files are located.
+
+    num_orbits: float
+        Number orbits from the start of the simulation. This function estimates
+        the duration of `num_orbits` from orbital phase which is obtained from
+        the coordinates of the black holes stored in the `Horizons.h5` file.
+
+    deltaTOverM: float, default=1
+        Time step in dimensionless units. It is used to interpolate the data
+        since the NR data in `Horizons.h5` are not stored at uniform time grid.
+
+    Returns
+    -------
+    num_obits_duration : float
+        Duration of first `num_orbits`.
+    """
+    horizons_data = h5py.File(horizon_filepath, "r")
+    xA_data = horizons_data["AhA.dir"]["CoordCenterInertial.dat"]
+    xB_data = horizons_data["AhB.dir"]["CoordCenterInertial.dat"]
+    time = xA_data[:, 0]
+    separion_vec = xA_data[:, 1:] - xB_data[:, 1:]
+    # We will assume the system to be non-precessing so that the motion
+    # is restricted in the x-y plane and the z-coordinate is effectively zero.
+    # Check if it is true, i.e., z-coordinate remains very small
+    if any(np.abs(separion_vec[:, 2]) > 1e-5):
+        raise Exception(
+            "The system seems to be precessing. Cannot get the orbital "
+            "phase by assuming the motion is restricted in the x-y plane.")
+    # Get the orbital phase
+    phase_orb = np.unwrap(np.arctan2(separion_vec[:, 1], separion_vec[:, 0]))
+    # Interpolate phase
+    time_interp = np.arange(min(time), max(time), deltaTOverM)
+    phase_interp = InterpolatedUnivariateSpline(time, phase_orb)(time_interp)
```

Comment:
I don't get it, why do you need to interpolate here?

Reply:
Because the NR times are not uniform. Maybe it would not make a huge difference but don't know how large the deltaT in NR files could be, so thought it would be safe to use interpolation.

Reply:
Nah, uniform times doesn't matter here. You just want the orbital time scale, and the time step will never be so large that the binaries jump a huge fraction of an orbit.

Reply:
Okay, removed that.